Source/OWASP-2013-Demo.Domain.Tests/UserProvider.Tests.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System.Web;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Moq;
+using OWASP_2013_Demo.Domain;
+using OWASP_2013_Demo.Interfaces.Entities;
+using OWASP_2013_Demo.Interfaces.Repositories;
+using OWASP_2013_Demo.Models.DB;
+
+namespace given_that_i_make_a_user_authenication_request
+{
+	[TestClass]
+	public class when_I_dont_apply_best_security_practices
+	{
+		private static UserProvider _userProvider;
+		private static Mock<HttpResponseBase> _mockedHttpResponseBase;
+
+		[ClassInitialize]
+		public static void Setup(TestContext testContext)
+		{
+			var mockedUserRepository = new Mock<IUserRepository>();
+			var mockedSiteConfiguration = new Mock<ISiteConfiguration>();
+			_mockedHttpResponseBase = new Mock<HttpResponseBase>();
+
+			mockedSiteConfiguration.Setup(x => x.SecureMode).Returns(false);
+
+			mockedUserRepository.Setup(x => x.FetchUserByEmailAddress("nouser@madeup.com"))
+				.Returns((IUser)null);
+
+			mockedUserRepository.Setup(x => x.FetchUserByEmailAddress("margaret0@adventure-works.com"))
+				.Returns(new User()
+				{
+					EmailAddress = "margaret0@adventure-works.com",
+					PasswordHash = "HyeF+GbkROa/eaUyYgVqCm8zMdNn/AEIzOnd+luTsgQ=",
+					PasswordSalt = "i2U3DxA="
+				});
+
+			// In the future maybe mock PasswordManager too.
+			_userProvider = new UserProvider(mockedUserRepository.Object, new PasswordManager(), 
```

Comment:
We're likely want to swap out the concrete implementation of PasswordManager here for a mock of IPasswordManager.